pkg/iac/ignore/parse.go

```diff
@@ -103,16 +113,16 @@ func parseComment(input string, rng types.Range, parsers []RuleSectionParser) (R
 			}
 
 			if parser.Parse(val) {
-				rule.sections[parser.Key()] = parser.Param()
+				sections[parser.Key()] = parser.Param()
 			}
 		}
 	}
 
-	if _, exists := rule.sections["id"]; !exists {
-		return Rule{}, errors.New("rule section with the `ignore` key is required")
+	if _, exists := sections["id"]; !exists {
+		return nil, errors.New("rule section with the `ignore` key is required")
```

Comment:
Are there any callers of this func that we need to add a check for when we return `nil` here? I only see [this](https://github.com/aquasecurity/trivy/blob/f6f2375c9e77bdcd8b1568accb1a387be9df31e0/pkg/iac/ignore/rule.go#L68) and it seems like this is covered.

Reply:
if comment parsing fails, we skip it https://github.com/aquasecurity/trivy/pull/6708/files#diff-c1fe13f24e97f852991501a2535948a2ff13ef4e36564313e85754d8f9c6f35eR58-R60

Reply:
Fair enough. I guess it is subjective as I only tend to return `nil` for structs that are passed around by reference. But I'm fine with this.